static/app/views/performance/browser/webVitals/utils/queries/storedScoreQueries/useTransactionSamplesWebVitalsScoresQuery.tsx

```diff
@@ -0,0 +1,135 @@
+  });
+
+  const eventView = EventView.fromNewQueryWithPageFilters(
+    {
+      fields: [
+        'id',
+        'user.display',
+        'transaction',
+        'measurements.lcp',
+        'measurements.fcp',
+        'measurements.cls',
+        'measurements.ttfb',
+        'measurements.fid',
+        'transaction.duration',
+        'replayId',
+        'timestamp',
+        'profile.id',
+        'project',
+        'measurements.score.total',
+        ...(webVital
+          ? [`measurements.score.${webVital}`, `measurements.score.weight.${webVital}`]
+          : []),
+      ],
+      name: 'Web Vitals',
+      query: `transaction.op:pageload transaction:"${transaction}" has:measurements.score.total ${
+        query ? query : ''
+      }`,
+      orderby: mapWebVitalToOrderBy(orderBy) ?? withProfiles ? '-profile.id' : undefined,
+      version: 2,
+    },
+    pageFilters.selection
+  );
+
+  eventView.sorts = [sort];
+
+  const {data, isLoading, ...rest} = useDiscoverQuery({
+    eventView,
+    limit: limit ?? 50,
+    location,
+    orgSlug: organization.slug,
+    options: {
+      enabled,
+      refetchOnWindowFocus: false,
+    },
+    referrer: 'api.performance.browser.web-vitals.transaction',
+  });
+
+  const toNumber = (item: ReactText) => (item ? parseFloat(item.toString()) : null);
+  const tableData: TransactionSampleRowWithScore[] =
+    !isLoading && data?.data.length
+      ? (data.data.map(
+          row => ({
+            id: row.id?.toString(),
+            'user.display': row['user.display']?.toString(),
+            transaction: row.transaction?.toString(),
+            'measurements.lcp': toNumber(row['measurements.lcp']),
+            'measurements.fcp': toNumber(row['measurements.fcp']),
+            'measurements.cls': toNumber(row['measurements.cls']),
+            'measurements.ttfb': toNumber(row['measurements.ttfb']),
+            'measurements.fid': toNumber(row['measurements.fid']),
+            'transaction.duration': toNumber(row['transaction.duration']),
+            replayId: row.replayId?.toString(),
+            'profile.id': row['profile.id']?.toString(),
+            projectSlug: row.project?.toString(),
+            timestamp: row.timestamp?.toString(),
+            score: Math.round((toNumber(row['measurements.score.total']) ?? 0) * 100),
+            ...(webVital
+              ? {
+                  [`${webVital}Score`]: Math.round(
+                    ((toNumber(row[`measurements.score.${webVital}`]) ?? 0) /
+                      (toNumber(row[`measurements.score.weight.${webVital}`]) ?? 0)) *
+                      100
+                  ),
+                  [`${webVital}Weight`]: Math.round(
+                    (toNumber(row[`measurements.score.weight.${webVital}`]) ?? 0) * 100
+                  ),
+                }
+              : {}),
+          })
+          // TODO: Discover doesn't let us query more than 20 fields and we're hitting that limit.
```

Comment:
Or we can change that limit 🤷, is that a pre-metrics or metrics constraint?

Reply:
I _think_ it's an artificial constraint in discover to proactively reduce load to clickhouse.
I don't actually need to query that many fields for now, but yeah we can look into changing the limit if it becomes a blocker.